### PR TITLE
Improve YaruPageIndicator

### PR DESCRIPTION
**Pull request by Jupi007**
No design change.

This PR allows various customization options, like:
- Change items size ;
- Use a custom item widget ;
- Use various layout spacing delegates ;
- Change the text indicator text style ;
- Use a custom text widget.

See also https://github.com/canonical/ubuntu-desktop-installer/issues/1348

[Capture vidéo du 2023-03-09 12-34-21.webm](https://user-images.githubusercontent.com/36476595/224011519-70147a5d-d2ab-4e88-9a09-2a86d72e0d3c.webm)

### CODE REVIEW

File: lib/src/widgets/yaru_page_indicator_layout_delegate.dart [hunk 1]
```diff
@@ -0,0 +1,109 @@
+import 'yaru_page_indicator.dart';
+
+/// Layout delegate interface which controls a [YaruPageIndicator] items spacing.
+abstract class YaruPageIndicatorLayoutDelegate {
+  double? calculatePaneWidth({
```
Comment (jpnurmi) [inline lib/src/widgets/yaru_page_indicator_layout_delegate.dart:5]: pane? item? :D

File: lib/src/widgets/yaru_page_indicator_layout_delegate.dart [hunk 2]
```diff
@@ -0,0 +1,109 @@
...
+    required double itemsWidth,
```
Comment (jpnurmi) [inline lib/src/widgets/yaru_page_indicator_layout_delegate.dart:6]: it is the width of a single item or the total width of all items?

File: lib/src/widgets/yaru_page_indicator_layout_delegate.dart (hunk 1 above)
Comment (Jupi007) [reply lib/src/widgets/yaru_page_indicator_layout_delegate.dart]: 🤦‍♂️ Copy-paste typo, fixed 👍

File: lib/src/widgets/yaru_page_indicator_layout_delegate.dart (hunk 2 above)
Comment (Jupi007) [reply lib/src/widgets/yaru_page_indicator_layout_delegate.dart:6]: It is the width of all items, maybe rename it in `allItemsWidth`?

Comment (jpnurmi) [reply lib/src/widgets/yaru_page_indicator_layout_delegate.dart]: Sounds good. I wasn't 100% sure if it was a typo or if it mean them all. :)

**Review by jpnurmi — APPROVED** [review]
LGTM and @d-loose even took it for a spin already! :+1:

**Pull request merged**